Composer/packages/server/assets/projects/ToDoBotWithLuisSample/dialogs/AddItem/AddItem.dialog

```diff
@@ -0,0 +1,114 @@
+{
+  "$type": "Microsoft.AdaptiveDialog",
+  "$designer": {
+    "name": "AddItem",
+    "id": "225905"
+  },
+  "autoEndDialog": "true",
+  "defaultResultProperty": "dialog.result",
+  "triggers": [
+    {
+      "$type": "Microsoft.OnBeginDialog",
+      "$designer": {
+        "name": "BeginDialog",
+        "id": "479346"
+      },
+      "actions": [
+        {
+          "$type": "Microsoft.SetProperties",
+          "$designer": {
+            "id": "811190",
+            "name": "Set properties"
+          },
+          "assignments": [
+            {
+              "property": "dialog.itemTitle",
+              "value": "=coalesce(@itemTite, $itemTitle)"
```

Comment:
Do we need coalesce $itemTitle here? I have tested and find that dialog.xx variables will always be null when the dialog is created.

Reply:
@xieofxie this is good practice to show in our samples that help users understand how to build sub-dialogs that can be composed easily with other dialogs. This sets the stage up for a sub-dialog to be called in both an intent trigger as well as for any re-use invocation where some properties might be passed in via `Options` to BeginDialog